internal/content_coding.go

```diff
@@ -370,6 +395,23 @@ func (d *ZlibDecoder) Decode(data []byte, maxDecompressionSize int64) ([]byte, e
 	return d.buf.Bytes(), nil
 }
 
+type ZstdDecoder struct {
+	decoder *zstd.Decoder
+}
+
+func NewZstdDecoder() (*ZstdDecoder, error) {
+	d, err := zstd.NewReader(nil, zstd.WithDecoderConcurrency(0))
+	return &ZstdDecoder{
+		decoder: d,
+	}, err
+}
+
+func (*ZstdDecoder) SetEncoding(string) {}
+
+func (d *ZstdDecoder) Decode(data []byte, _ int64) ([]byte, error) {
+	return d.decoder.DecodeAll(data, nil)
+}
```

Comment:
Return error when decoded data is larger than `maxDecompressionSize`. See `func (*IdentityDecoder) Decode` or others..